R/aggregate_by_chi.R

```diff
@@ -203,11 +203,11 @@ aggregate_ch_episodes <- function(episode_file) {
   data.table::setDT(episode_file)
 
   # Perform grouping and aggregation
-  episode_file <- episode_file[, `:=`(
-    ch_no_cost = max(ch_no_cost),
-    ch_ep_start = min(record_keydate1),
-    ch_ep_end = max(ch_ep_end),
-    ch_cost_per_day = mean(ch_cost_per_day)
+  episode_file[, c(
+    ch_no_cost := max(ch_no_cost),
+    ch_ep_start := min(record_keydate1),
+    ch_ep_end := max(ch_ep_end),
+    ch_cost_per_day := mean(ch_cost_per_day)
   ), by = c("chi", "ch_chi_cis")]
```

Comment:
This does fix the build warning but I don't think it works. I ran it against some dummy data and it wasn't working (but the original code was).

I looked at `help(":=", "data.table")` and then tested this code and I think this is a replacement that works and also doesn't trigger a build warning.
```suggestion
episode_file[, c(
    "ch_no_cost",
    "ch_ep_start",
    "ch_ep_end",
    "ch_cost_per_day"
  ) := list(
    max(ch_no_cost),
    min(record_keydate1),
    max(ch_ep_end),
    mean(ch_cost_per_day)
  ),
  by = c("chi", "ch_chi_cis")]
```

Could you have a check please Zihao?

Reply:
Many thanks for this! I tested this way using dummy data and it works perfectly.